### PR TITLE
MV3: Migrate permissions

### LINKED ISSUE #15769: [MV3]: Permissions migration from MV2 to MV3
With the migration to MV3, permissions have now been divided into 3 categories: permissions, optional_permissions, and host_permissions. This PR performs the MV3 migration for permissions.

### References:

https://developer.chrome.com/docs/extensions/mv3/declare_permissions/
https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#host-permissions

### More Information:

https://github.com/MetaMask/metamask-extension/pull/15448/files

### Current permission errors shown when running MV3: 

image

### Steps to reproduce

- Open "Errors" console of the extension
- Verify permissions continue to work as expected

### Error messages or log output

_No response_

### Version

14.18.1

### Build type

_No response_

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

For info, see: https://github.com/MetaMask/metamask-extension/issues/15769 

## More Information

* Fixes https://github.com/MetaMask/metamask-extension/issues/15769

### Before

Screenshot 2022-09-06 at 3 47 39 AM

### After

Screenshot 2022-09-09 at 3 25 31 AM

## Manual Testing Steps

1. Load the MV3 extension build
2. Go to chrome://extensions/
3. Click "Errors"
4. Observe no URL errors are no longer shown (note that other errors might appear if you are running the extension and start the service worker)

## Pre-Merge Checklist

- [ ] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [ ] PR is linked to the appropriate GitHub issue
- [ ] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

### CODE REVIEW

**Review by jpuri — DISMISSED** [review]
Nice work @digiwand 👏

**Review by 0ex-d — DISMISSED** [review]

File: app/manifest/v3/_base.json [hunk 1]
```diff
@@ -47,7 +47,16 @@
   ],
   "default_locale": "en",
   "description": "__MSG_appDescription__",
-  "host_permissions": ["file://*/*", "http://*/*", "https://*/*"],
+  "host_permissions": [
+    "http://localhost:8545/",
+    "file://*/*",
+    "http://*/*",
+    "https://*/*",
+    "*://*.eth/",
```
Comment (digiwand) [inline app/manifest/v3/_base.json:55]: I don't think we need this or the 3 URLs following after we are using 
```
    "http://*/*",
    "https://*/*",
```

keeping these specific URLs to be more explicit

Comment (jpuri) [reply app/manifest/v3/_base.json:55]: This can be complete list
```
    "file://*/*",
    "http://*/*",
    "https://*/*",
    "*://*.eth/",
```

Comment (digiwand) [reply app/manifest/v3/_base.json]: ok 👍  https://github.com/MetaMask/metamask-extension/pull/15777/commits/4721af8c9de169c0f208d5d5d4b1babb7f2e06d0

Comment (Gudahtt) [reply app/manifest/v3/_base.json]: We'll need to restore those URLs once we drop the wildcard host permissions that we needed just for dynamic contentscript injection. We won't need that once the Chromium team fixes MAIN-world contentscript support.

But dropping them makes sense for now

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by jpuri — APPROVED** [review]

**Comment by digiwand** [comment]
Once the Chromium team fixes MAIN-world contentscript support, we will remove the wildcard URLs, and we will want to restore 
```
    "*://*.eth/",
    "https://*.infura.io/",
    "https://chainid.network/chains.json",
    "https://lattice.gridplus.io/*"
```

see comment: https://github.com/MetaMask/metamask-extension/pull/15777#discussion_r969528917, 
see commit: 4721af8c9de169c0f208d5d5d4b1babb7f2e06d0

**Pull request merged**